ante-ls/src/main.rs

```diff
@@ -196,40 +162,26 @@ impl Backend {
             });
 
             let loc = diagnostic.locate();
-            let filename = root.join(loc.filename);
-            let filename = match filename.canonicalize() {
-                Ok(filename) => filename,
-                Err(_) => {
-                    self.client
-                        .log_message(
-                            MessageType::ERROR,
-                            format!("Diagnostics for file {filename:?}, but its path could not be canonicalized"),
-                        )
-                        .await;
-                    continue;
-                },
-            };
-            let uri = Url::from_file_path(filename).unwrap();
+            let uri = Url::from_file_path(loc.filename).unwrap();
 
             let rope = match self.document_map.get(&uri) {
                 Some(rope) => rope,
                 None => {
-                    // Can we somehow retrieve the file from the compiler rather than reading it again?
-                    // Or have the compiler go through the lsp server file buffer instead of reading it from the file system?
-                    let rope = Rope::from_str(&std::fs::read_to_string(uri.path()).unwrap());
-                    self.document_map.insert(uri.clone(), rope.clone());
+                    let contents = cached_read(&cache.file_cache, loc.filename).unwrap();
+                    let rope = Rope::from_str(&contents);
+                    self.document_map.insert(uri.clone(), rope);
                     self.document_map.get(&uri).unwrap()
                 },
             };
 
-            // TODO: This range seems to be off
```

Comment:
What exactly was off about it? Not sure if it's just fixed now because they seem fine to me:

<details>

<img width="2441" alt="image" src="https://github.com/jfecher/ante/assets/20847625/f4089737-d1ab-42d0-aff1-17d80b5df7b3">

</details>

Reply:
I have an example I'm working on currently where the location is off:
![image](https://github.com/jfecher/ante/assets/13188781/f79d1e3f-7772-4ccb-b3e5-19db493f9f55)

Running the compiler on the file reports:
![image](https://github.com/jfecher/ante/assets/13188781/01d54cea-e7b6-4cb4-b0a3-47dc91c39b8f)

Where the line and column info are correct.

Reply:
You can ignore the contents of the file. I've found a good way of making Ante thread-safe while also allowing aliased mutability without lifetime annotations!

Reply:
Ok, I've determined this is because of larger unicode characters present in the file causing the internal byte index to be different than the character index.